src/test/integration/services/userService.spec.js

```diff
@@ -0,0 +1,110 @@
+const mongoose = require('mongoose')
+const { MongoMemoryServer } = require('mongodb-memory-server')
+const userService = require('~/services/user')
+const User = require('~/models/user')
+const { createError } = require('~/utils/errorsHelper')
+const { DOCUMENT_NOT_FOUND, ALREADY_REGISTERED } = require('~/consts/errors')
+
+let mongoServer
+
+beforeAll(async () => {
+  mongoServer = await MongoMemoryServer.create()
+  const uri = mongoServer.getUri()
+  await mongoose.connect(uri, { useNewUrlParser: true, useUnifiedTopology: true })
+})
+
+afterAll(async () => {
+  await mongoose.disconnect()
+  await mongoServer.stop()
+})
+
+beforeEach(async () => {
+  await User.deleteMany({})
+})
+
+describe('User Service', () => {
+  describe('createUser', () => {
+    it('should create a new user successfully', async () => {
+      const user = await userService.createUser('student', 'Eva', 'Mendez', 'em@gmailcom', 'qwerty9876', 'en')
+
+      const createdUser = await User.findById(user._id).lean().exec()
+      expect(createdUser).toBeDefined()
+      expect(createdUser.email).toBe('em@gmailcom')
+      expect(createdUser.firstName).toBe('Eva')
+      expect(createdUser.lastName).toBe('Mendez')
+    })
+
+    it('should throw ALREADY_REGISTERED error if user already exists', async () => {
+      await userService.createUser('student', 'Eva', 'Mendez', 'em@gmailcom', 'qwerty9876', 'en')
+
+      await expect(
+        userService.createUser('student', 'Eva', 'Mendez', 'em@gmailcom', 'qwerty9876', 'en')
+      ).rejects.toThrowError(createError(409, ALREADY_REGISTERED))
+    })
+  })
+
+  describe('getUserById', () => {
+    it('should retrieve a user by id', async () => {
+      const user = new User({
+        _id: new mongoose.Types.ObjectId(),
+        firstName: 'Eva',
+        lastName: 'Mendez',
+        email: 'em@gmailcom',
+        password: 'qwerty9876'
+      })
+      await user.save()
+
+      const retrievedUser = await userService.getUserById(user._id)
+      expect(retrievedUser).not.toBeNull()
+      expect(retrievedUser._id.toString()).toEqual(user._id.toString())
```

Comment:
Is it necessary to check the string variant of the ObjectId here?
Maybe you can just use
```js
expect(retrievedUser._id).toEqual(user._id)
```